File: tasks/prac8_python/task5.py
def analyze_text(text):
    """
    Анализирует текст и возвращает статистику

    Args:
        text: строка для анализа

    Returns:
        Словарь со статистикой текста
    """
    # Убираем лишние пробелы в начале и конце
    text = text.strip()

    # Разбиваем текст на предложения по точкам
    # filter(None, ...) убирает пустые строки из результата
    sentences = [s.strip() for s in text.split('.') if s.strip()]

    # Разбиваем текст на слова по пробелам
    words = text.split()

    # Находим самую длинную и короткую фразу
    # max() и min() с key=len
    # находят элементы с максимальной/минимальной длиной
    longest_sentence = max(sentences, key=len) if sentences else ""
    shortest_sentence = min(sentences, key=len) if sentences else ""

    # Возвращаем словарь со всей статистикой
    return {
        'symbols': len(text),           # Общее количество символов
        'words': len(words),            # Количество слов
        'sentences': len(sentences),    # Количество предложений
        'longest': longest_sentence,    # Самое длинное предложение
        'shortest': shortest_sentence   # Самое короткое предложение
    }


if __name__ == "__main__":
    print("=== Анализатор текста ===\n")

    test_text = "Привет! Как дела? Все хорошо. \
        Это тестовый текст для проверки работы анализатора."

    print("Анализируемый текст:")
    print(test_text)
    print()

    stats = analyze_text(test_text)

    print("Результаты анализа:")
    print(f"Символов: {stats['symbols']}")
    print(f"Слов: {stats['words']}")
    print(f"Предложений: {stats['sentences']}")
    print(f"Самое длинное предложение: '{stats['longest']}'")
    print(f"Самое короткое предложение: '{stats['shortest']}'")

    # Дополнительный тест с пустым текстом
    print("\nТест с пустым текстом:")
    empty_stats = analyze_text("")
    print(empty_stats)

File: tasks/prac8_js/task5.js
const prompt = require('prompt-sync')();

function analyzeText(text) {
    /**
     * Анализирует текст и возвращает статистику
     * @param {string} text - строка для анализа
     * @returns {Object} Объект со статистикой текста
     */
    // Убираем лишние пробелы в начале и конце
    text = text.trim();
    
    // Разбиваем текст на предложения по точкам, восклицательным и вопросительным знакам
    // filter(Boolean) убирает пустые строки из результата
    const sentences = text.split(/[.!?]+/).filter(sentence => 
        sentence.trim().length > 0
    ).map(sentence => sentence.trim());
    
    // Разбиваем текст на слова по пробелам и знакам препинания
    const words = text.split(/\s+/).filter(word => word.length > 0);
    
    // Находим самую длинную и короткую фразу
    // reduce() находит элементы с максимальной/минимальной длиной
    const longestSentence = sentences.reduce((longest, current) => 
        current.length > longest.length ? current : longest, ""
    );
    
    const shortestSentence = sentences.reduce((shortest, current) => 
        current.length < shortest.length || shortest === "" ? current : shortest, ""
    );
    
    // Возвращаем объект со всей статистикой
    return {
        symbols: text.length,           // Общее количество символов
        words: words.length,            // Количество слов
        sentences: sentences.length,    // Количество предложений
        longest: longestSentence,       // Самое длинное предложение
        shortest: shortestSentence      // Самое короткое предложение
    };
}

// Демонстрация работы
console.log("=== Анализатор текста ===\n");

const testText = "Привет! Как дела? Все хорошо. Это тестовый текст для проверки работы анализатора.";

console.log("Анализируемый текст:");
console.log(testText);
console.log();

const stats = analyzeText(testText);

console.log("Результаты анализа:");
console.log("Символов:", stats.symbols);
console.log("Слов:", stats.words);
console.log("Предложений:", stats.sentences);
console.log("Самое длинное предложение:", `'${stats.longest}'`);
console.log("Самое короткое предложение:", `'${stats.shortest}'`);

// Дополнительный тест с пустым текстом
console.log("\nТест с пустым текстом:");
const emptyStats = analyzeText("");
console.log(emptyStats);